wsd/FileServer.cpp

```diff
@@ -494,24 +490,20 @@ void FileServerRequestHandler::sendError(int errorCode, const Poco::Net::HTTPReq
                                          const std::string& shortMessage, const std::string& longMessage,
                                          const std::string& extraHeader)
 {
-    Poco::URI requestUri(request.getURI());
-    const std::string& path = requestUri.getPath();
-    std::ostringstream oss;
-    oss << "HTTP/1.1 " << errorCode << "\r\n"
-        "Content-Type: text/html charset=UTF-8\r\n"
-        "Date: " << Util::getHttpTimeNow() << "\r\n"
-        "User-Agent: " << WOPI_AGENT_STRING << "\r\n"
-        << extraHeader
-        << "\r\n";
+    std::string body = "";
+    std::string headers = extraHeader;
     if (!shortMessage.empty())
     {
+        Poco::URI requestUri(request.getURI());
+        const std::string& path = requestUri.getPath();
```

Comment:
Any good reason to use & here? As path is used only once (to encode) it's better to simply pass requestUri.getPath() to the encode function.

Reply:
Just that that was the original code from  https://github.com/CollaboraOnline/online/blob/1ef9ad4f5688ee97468aeea32e895b11333dd8ef/wsd/FileServer.cpp#L498 . But, sure, changed in 3d9ee4292